### PR TITLE
added support for openshift

### PR DESCRIPTION
**Pull request by cameronbraid**

### CODE REVIEW

**Comment by jbadeau** [comment]
Is this going to be merged? Or whats the plan?

File: reportportal/v5/templates/openshift-gateway-deployment.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  labels: {{ include "labels" . | indent 4  }}
+  name: {{ $fullName }}-gateway
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      component: {{ $fullName }}-gateway
+  strategy:
+    rollingUpdate:
+      maxSurge: 25%
```
Comment (avarabyeu) [inline reportportal/v5/templates/openshift-gateway-deployment.yml]: can we have those values moved to values.yaml?

File: reportportal/v5/templates/openshift-gateway-deployment.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      maxUnavailable: 25%
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        component: {{ $fullName }}-gateway
+    spec:
+      containers:
+        - image: 'traefik:v2.0'
```
Comment (avarabyeu) [inline reportportal/v5/templates/openshift-gateway-deployment.yml]: can we have this value moved to values.yaml?

File: reportportal/v5/templates/openshift-gateway-deployment.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+          imagePullPolicy: IfNotPresent
+          name: traefik
+          ports:
+            - containerPort: 8080
```
Comment (avarabyeu) [inline reportportal/v5/templates/openshift-gateway-deployment.yml]: can we have this value moved to values.yaml?

File: reportportal/v5/templates/openshift-gateway-route.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  host: {{ $host }}
+  port:
+    targetPort: http
+  tls:
+    insecureEdgeTerminationPolicy: Redirect
+    termination: edge
+  to:
+    kind: Service
+    name: {{ $fullName }}-gateway
+    weight: 100
```
Comment (avarabyeu) [inline reportportal/v5/templates/openshift-gateway-route.yml]: can we have this value moved to values.yaml?

File: reportportal/v5/templates/migrations-job.yaml [hunk 5]
```diff
@@ -9,6 +9,9 @@ spec:
       labels:
         component: {{ include "reportportal.fullname" . }}-migrations
     spec:
+      {{- if .Values.openshift.enable }}
+      serviceAccountName: {{ template "reportportal.serviceAccountName" . }}
```
Comment (avarabyeu) [inline reportportal/v5/templates/migrations-job.yaml]: this service account has additional permissions to read some state over k8s api. Do we really this for openshift?

File: reportportal/v5/templates/openshift-gateway-configmap.yml [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+            rule: "PathPrefix(`/api`)"
+            middlewares:
+            - strip-api
+            service: "api"
+        
+        services:
+          uat:
+            loadBalancer:
+              servers:
+              - url: "http://reportportal-uat:9999/"
```
Comment (waynesun09) [inline reportportal/v5/templates/openshift-gateway-configmap.yml]: the uat url hostname should also use reportportal full name {{ $fullName }}, so it should be updated to:

url: "http://{{ $fullName }}-uat:9999"

File: reportportal/v5/templates/openshift-gateway-configmap.yml [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        
+          index:
+            loadBalancer:
+              servers:
+              - url: "http://reportportal-index:8080/"
```
Comment (waynesun09) [inline reportportal/v5/templates/openshift-gateway-configmap.yml]: also here with index hostname

File: reportportal/v5/templates/openshift-gateway-configmap.yml [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+        
+          api:
+            loadBalancer:
+              servers:
+              - url: "http://reportportal-api:8585/"
```
Comment (waynesun09) [inline reportportal/v5/templates/openshift-gateway-configmap.yml]: same here

File: reportportal/v5/templates/openshift-gateway-configmap.yml [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+        
+          ui:
+            loadBalancer:
+              servers:
+              - url: "http://reportportal-ui:8080/"
```
Comment (waynesun09) [inline reportportal/v5/templates/openshift-gateway-configmap.yml]: same here

**Comment by shiva4983** [comment]
Hi, Please let me know if these changes are already in place?

**Pull request closed without merging**